src/native/corehost/hostmisc/utils.cpp

```diff
@@ -358,12 +408,18 @@ bool get_dotnet_root_from_env(pal::string_t* dotnet_root_env_var_name, pal::stri
         return true;
 
 #if defined(WIN32)
-    if (pal::is_running_in_wow64())
+    if (is_running_in_wow64())
     {
         *dotnet_root_env_var_name = _X("DOTNET_ROOT(x86)");
         if (get_file_path_from_env(dotnet_root_env_var_name->c_str(), recv))
             return true;
     }
+    else if (is_emulating_x64())
```

Comment:
After closer look, I do not understand why this case is needed at all. It should be taken care of by the `append(to_upper(get_arch()))` case above. 

Do we have a repro or test that demonstrates what this is fixing, and that is failing before this change and passing with this change?